### ISSUE TITLE
FCM Integration and sending Images to User. 

### ISSUE THREAD

**Issue by puneet739** [puneet739/SocietyFocus-Android#8]
We should be able to send promotional Images to user. 
Also bday reminder notification should be sent to user from backend.

**Comment by puneet739** [comment]
On clicking on notification, The app should get open with a new activity to display the image provided. 

on clicking back, user would be landed back on home page of app.

**Comment by puneet739** [comment]
@laherajmani This is under development.. 
How can i send multiple notification to user. Ie 2 different notification. 
On clicking the notification it opens a blank page. 

Expected flow--> If image is available --> Open a page with Image showing else open home page.

**Comment by SagarKhurana** [comment]
@puneet739 the flow is complete... kindly test it in recent commit i am just going to do.

**Comment by puneet739** [comment]
@SagarKhurana  This one is not landing to user image page.. Landing to blank page. Kindly check that once.

**Comment by puneet739** [comment]
@SagarKhurana  This is not done yet

**Comment by SagarKhurana** [comment]
@puneet739 
hi Puneet, it is working from my end and tested. please let me know where you are facing the issue.

Kindly merge the new pull request made by me and test the code.

**Comment by SagarKhurana** [comment]
@puneet739  
below is the format to send the data from postman

{
    "data": 
    {
        "body" : "Good",
        "title" : "Perfect",
        "icon" : "myicon"
        "main_picture" :"Image_URL"
    },
    "to" : "FCM DeviceID"
  }

in Header add below

Authentication : "key='yourserverkey_from_fcm_console'"

please take reference from the link https://developers.google.com/cloud-messaging/http

**Comment by puneet739** [comment]
@SagarKhurana  and @laherajmani  This is working fine but only with single message. 
If i am sending 2 push message. First message is overridden by second. Kindly check that once.

**Comment by SagarKhurana** [comment]
@puneet739  yes you are right the new message will override the older message... but u must have noticed each time u send notification u get a notification sound/vibration on ur phone as if new notification has arrived. 
it is because we cannot show two notifications from single application in diff. UI... what whatsapp and other app does they add their new message in the content of old notification and then send it again. 
thats y u see 2 new messages, 3 new messages... Hope you understood my point.

**Comment by puneet739** [comment]
@SagarKhurana  What watsapp does is to add the content in same notification. We are asking to send new notification everytime. Like Lenskart or anFlipkart, they send different different promotion on app. This is something we are looking here.

**Comment by SagarKhurana** [comment]
@puneet739  the desired feature has been updated... every time we would be getting a new message for each notification... but if we want to send the notification to previous or older message we need to add another variable in our data message that is notification_id, that will be provided in api and the id should be same. If we need different notification... we can send different id.

**Comment by puneet739** [comment]
Hi sagar,

In case you have committed some code, kindly push that.. So that i can take
the latest build.

Regards
Puneet Behl
0-9711616135

**Comment by puneet739** [comment]
@laherajmani Kindly look into this.

**Comment by laherajmani** [comment]
@puneet739 please check comment from sagar, change is required from api , add notification_id in data message
